test/gfx/test.sh

```diff
@@ -37,35 +39,30 @@ failTest () {
 	echo "${bold}${red}Test ${cmdline} failed!${1:+ (RC=$1)}${rescolors}${resbold}"
 }
 
-
-# Draw a random tile offset and VRAM0 size
-# Neither should change anything to how the image is displayed
-while [[ "$ofs" -eq 0 ]]; do (( ofs = RANDOM % 256 )); done
-while [[ "$size" -eq 0 ]]; do (( size = RANDOM % 256 )); done
-for f in *.bin; do
-	for flags in ""{," -b $ofs"}{," -N $size,256"}; do
-		newTest ./rgbgfx_test "$f" $flags
-		runTest || failTest $?
-	done
-done
-
-# Test round-tripping '-r' with '-c #none'
-reverse_cmd="$RGBGFX -c#none,#fff,#000 -o none_round_trip.2bpp -r 1 out.png"
-reconvert_cmd="$RGBGFX -c#none,#fff,#000 -o result.2bpp out.png"
-compare_cmd="cmp none_round_trip.2bpp result.2bpp"
-newTest "$reverse_cmd && $reconvert_cmd && $compare_cmd"
-runTest || failTest $?
-
-# Remove temporaries (also ignored by Git) created by the above tests
-rm -f out*.png result.png result.2bpp
-
 for f in *.png; do
 	flags="$([[ -e "${f%.png}.flags" ]] && echo "@${f%.png}.flags")"
 
+	desired_output=
+	if [[ -e "${f%.png}.out.bin" ]]; then
+		desired_output="${f%.png}.out.bin"
+		flags="$flags -o $outtmp"
+	elif [[ -e "${f%.png}.out.tilemap" ]]; then
+		desired_output="${f%.png}.out.tilemap"
+		flags="$flags -t $outtmp"
+	elif [[ -e "${f%.png}.out.attrmap" ]]; then
+		desired_output="${f%.png}.out.attrmap"
+		flags="$flags -a $outtmp"
+	elif [[ -e "${f%.png}.out.pal" ]]; then
+		desired_output="${f%.png}.out.pal"
+		flags="$flags -p $outtmp"
+	fi
+
 	newTest "$RGBGFX" $flags "$f"
```

Comment:
That would require N different `mktemp`s to allow for all possible outputs at once. I was thinking if we wanted to test multiple, we'd just copy the PNG; we already have some test cases like that, reusing the same images. (Plus, some of the outputs might expect to cause an error, while others should succeed.)

Reply:
Some of the behaviours may indeed be testable by running each flag individually, but sometimes we want to test the *combination* of such flags. (Plus it would reduce the number of test cases, and thus the number of files within the directory.)

The file names are encoded statically as `result.<ext>`, as you an see on line 50; this avoids the need for a bunch of `mktemp`s. They're already `.gitignore`d from the “seed” tests.

Reply:
Yeah, I added that line. :P I had wanted to limit the result.\* files to the `rgbgfx_test` seed tests, but sure, that makes sense.